src/Compilers/CSharp/Portable/Binder/Binder_Constraints.cs

```diff
@@ -86,34 +86,64 @@ internal partial class Binder
         /// <summary>
         /// Bind and return a single type parameter constraint clause.
         /// </summary>
-        private TypeParameterConstraintClause BindTypeParameterConstraints(string name, SeparatedSyntaxList<TypeParameterConstraintSyntax> constraintsSyntax, DiagnosticBag diagnostics)
+        private TypeParameterConstraintClause BindTypeParameterConstraints(
```

Comment:
Found: CS0449ERR_RefValBoundMustBeFirst, CS0451ERR_NewBoundWithVal, CS0401ERR_NewBoundMustBeLast and CS0706ERR_BadConstraintType.

These tests already called CreateCompilation(...).VerifyDiagnostics.  That's why we didn't see any affected tests.